src/state/routing/utils.ts

```diff
@@ -1,70 +1,95 @@
-import { Currency, CurrencyAmount, Token, TradeType } from '@uniswap/sdk-core'
+import { Currency, CurrencyAmount, Token } from '@uniswap/sdk-core'
 import { Pair, Route as V2Route } from '@uniswap/v2-sdk'
 import { FeeAmount, Pool, Route as V3Route } from '@uniswap/v3-sdk'
+import { nativeOnChain } from 'constants/tokens'
+import { PoolType } from 'hooks/routing/types'
+import { SwapRouterNativeAssets } from 'utils/currencyId'
 
-import { InterfaceTrade, QuoteResult, V2PoolInRoute, V3PoolInRoute } from './types'
+import { GetQuoteArgs, InterfaceTrade, QuoteResult, V2PoolInRoute, V3PoolInRoute } from './types'
 
 /**
- * Transforms a Routing API quote into an array of routes that can be used to create
- * a `Trade`.
+ * Transforms a Routing API quote into an array of routes that can be used to
+ * create a `Trade`.
  */
 export function computeRoutes(
-  currencyIn: Currency | undefined,
-  currencyOut: Currency | undefined,
-  tradeType: TradeType,
-  quoteResult: Pick<QuoteResult, 'route'> | undefined
-) {
-  if (!quoteResult || !quoteResult.route || !currencyIn || !currencyOut) return undefined
+  tokenInIsNative: boolean,
+  tokenOutIsNative: boolean,
+  quoteResult?: Pick<QuoteResult, 'route'>
+):
+  | {
+      routev3: V3Route<Currency, Currency> | null
+      routev2: V2Route<Currency, Currency> | null
+      inputAmount: CurrencyAmount<Currency>
+      outputAmount: CurrencyAmount<Currency>
+    }[]
+  | undefined {
+  if (!quoteResult || !quoteResult.route) return
 
   if (quoteResult.route.length === 0) return []
 
-  const parsedTokenIn = parseToken(quoteResult.route[0][0].tokenIn)
-  const parsedTokenOut = parseToken(quoteResult.route[0][quoteResult.route[0].length - 1].tokenOut)
+  const tokenIn = quoteResult.route[0]?.[0]?.tokenIn
+  const tokenOut = quoteResult.route[0]?.[quoteResult.route[0]?.length - 1]?.tokenOut
 
-  if (parsedTokenIn.address !== currencyIn.wrapped.address) return undefined
-  if (parsedTokenOut.address !== currencyOut.wrapped.address) return undefined
+  if (!tokenIn || !tokenOut) throw new Error('Expected both tokenIn and tokenOut to be present')
```

Comment:
do you want this to get caught by the `ErrorBoundary` and display the error message? if so, you should test this case. it might require you to use `useAsyncError` depending on how this is called

Reply:
depends, we throw errors in other parts of the quote flow too and those are suppressed by just returning `undefined` for a trade and console logging the error, so this is consistent with the existing flow

Reply:
i plan to address in a separate PR, but it would be nice to get to a point where we return all errors as part of rtk-query's default error handling mechanism, and we can access all errors through `const { data, error } = useGetQuoteQueryState` and then do something like this: https://github.com/Uniswap/mobile/blob/main/src/features/transactions/swap/useSwapWarnings.tsx#L58